### ISSUE TITLE
Heatmap: Don't count commits by other authors

### ISSUE THREAD

**Issue by timokoesters** [go-gitea/gitea#8178]
## Description
I noticed this when I started mirroring a public project. All mirrored commits count as a contribution from me on the heatmap even when someone else wrote the commit.

**Comment by sapk** [comment]
I think it is similar to #7285. In fact, Gitea doesn't attach the commit to you but create an activity of syncing mirror. Commits of mirrored repo are not parsed.

**Comment by timokoesters** [comment]
![Screenshot_20190916_180751](https://user-images.githubusercontent.com/25297359/64974370-f7825880-d8ac-11e9-9f7e-dddcded8478e.png)
Here's what it looks like.

**Comment by sapk** [comment]
Yes that what I mean. Currently you get a activity to the owner user each time gitea check the remote and there is a commit to fetch. In this case, the commit are not parsed like a push and are not attributed the commiter.

**Comment by timokoesters** [comment]
I'm still interested in this.

**Comment by silverwind** [comment]
Also noticed this. Contribution is attributed to the pusher, not the actual author (which is what one usually wants).

**Comment by timokoesters** [comment]
This feature is completely useless to me in it's current state

![Screenshot_20200331_202301](https://user-images.githubusercontent.com/25297359/78061379-72b0e500-738d-11ea-8be1-ec7a8309ca15.png)

**Comment by DiJuMx** [comment]
I would also like to see this be "Activity as dictated by commits". 
Sometimes I work offline for multiple days at a time, only remembering to push later. 
Presently, this means I get individual hot spots; whereas I would like to see how often I have committed.

Perhaps we could have a (site-wide) option which determines which tracking method is being used?
Or perhaps more configurability as to what counts? (For example if someone only cares about pull/merge requests accepted).

There is precedent for this; [as this (using commit date) is how Github operates.](https://docs.github.com/en/account-and-profile/setting-up-and-managing-your-github-profile/managing-contribution-settings-on-your-profile/viewing-contributions-on-your-profile#how-contribution-event-times-are-calculated)

**Comment by leaveye** [comment]
guess Gitea do not think this worth to response?

**Comment by Username404-59** [comment]
Yeah something should be fine about
As @timokoesters this makes the feature mostly useless

**Comment by Username404-59** [comment]
Yeah something should be done about it
It makes the feature basically useless, like @timokoesters said.

**Comment by lunny** [comment]
Let's clarify which behaviors should be counted.

- [ ] 👍 Your emails as a committer of the pushing commit
- [ ] 💯 Your emails as an author of the the pushing commit
- [ ] 🎉 You pushed the commits
- [ ] ❤️ Include mirror repositories
- [ ] 🥇 Include imported repositories
- [ ] Others

**Comment by leaveye** [comment]
any update on this? reviewing my heatmap reminds me this issue.
